src/main/java/bio/terra/service/common/azure/StorageTableName.java

```diff
@@ -0,0 +1,59 @@
+package bio.terra.service.common.azure;
+
+import bio.terra.common.exception.NotImplementedException;
+import java.security.InvalidParameterException;
+import java.util.UUID;
+
+public enum StorageTableName {
+  SNAPSHOT_TABLE("snapshot"),
+  LOAD_HISTORY_TABLE("loadHistory"),
+  DEPENDENCIES_TABLE("dependencies"),
+  DATASET_TABLE("dataset") {
+    // TODO - With DR-2127, move remove special case for dataset
+    @Override
+    public String toTableName(UUID resourceId) {
+      throw new NotImplementedException(
+          "Dataset storage table names are not unique per resource. Use DATASET_TABLE.toTableName() instead.");
+    }
+
+    public String toTableName() {
+      return this.label;
+    }
+  },
+  FILES_TABLE("files") {
+    @Override
+    public String toTableName(UUID resourceId) {
+      throw new NotImplementedException(
+          "Files storage table names are not unique per resource. Use FILES_TABLE.toTableName() instead.");
+    }
+
+    public String toTableName() {
+      return this.label;
+    }
+  };
+
+  public final String label;
+
+  StorageTableName(String label) {
+    this.label = label;
+  }
+
+  /**
+   * Generate a Storage Table name from a UUID
+   *
+   * @param resourceId The datasetId or snapshotId to be used as root of the table name
+   * @return A valid azure storage table name
+   */
+  public String toTableName(UUID resourceId) {
+    if (resourceId == null) {
+      throw new InvalidParameterException(
```

Comment:
I'm not sure `InvalidParameterException` is the best exception to use here. That error is part of the java cryptography architecture. For a general parameter check, I think `IllegalArgumentException` would be a better choice.
```suggestion
      throw new IllegalArgumentException(
```